src/main/kotlin/org/elm/lang/core/psi/elements/ElmTypeAnnotation.kt

```diff
@@ -54,6 +57,7 @@ class ElmTypeAnnotation(node: ASTNode) : ElmPsiElementImpl(node), ElmReferenceEl
     override val referenceName: String
         get() = referenceNameElement.text
 
-    override fun getReference() =
-            LocalTopLevelValueReference(this)
+    override fun getReference(): ElmReference =
+            if (parent is PsiFile) LocalTopLevelValueReference(this)
```

Comment:
Do I need to do anything different here to account for stubs?

Reply:
Nope, this is fine as-is. The Psi stubs only come into play for declarations that (1) introduce a name (e.g. a `PsiNamedElement`) and (2) are visible outside of the file in which it is defined (they can be imported)